CMakeLists.txt

```diff
@@ -0,0 +1,52 @@
+cmake_minimum_required(VERSION 3.1)
+
+project(BoostMp11 VERSION 1.66.0 LANGUAGES CXX)
+
+set(CMAKE_CXX_STANDARD 11)
+set(CMAKE_CXX_STANDARD_REQUIRED ON)
```

Comment:
Since mp11 can optionally use features from C++17 and does its own C++ feature detection it's probably better to set `CMAKE_CXX_STANDARD_REQUIRED` to `OFF` and set `CMAKE_CXX_STANDARD` into the max standard supported by CMake. I'd prefer to make it in a separate commit if possible.

Reply:
If you set `CMAKE_CXX_STANDARD` to the max know to CMake you will exclude a lot of compilers, such as NVCC, that do not yet have e.g. C++17 support.

Reply:
Yep. But I just realized that the C++ standard is really applied only to tests since the library itself is header-only. For tests it would be the best to test different language versions to make sure that mp11 compatibility layer works fine and library is compatible with all C++ versions (C++11, C++14 and C++17 for now). That's quite easy to add and I'll do it later in separate PR.

Reply:
Yes, in that case only set the requirement on the tests (per target) instead of putting it on the library target :)

Reply:
Let's keep it like it is for this PR. I'd prefer to move c++11 requirement to test targets in a separate commit.